service-catalog/gcp-backup/README.md

```diff
@@ -0,0 +1,132 @@
+
+1. _AWS Account compromise_. The [threat model] for Rust's infrastructure, created by the Rust Foundation's security engineer, highlights the risk of an AWS account compromise.
+   If a malicious actor was able to gain administrator access to the AWS account of one of the [infra-admins],
+   they could bypass a lot of safe guards and delete data.
+2. _AWS Account deletion_. AWS could accidentally delete our account, resulting in the possible deletion of data and backups.
+   For example, something similar happened at [Google](https://arstechnica.com/gadgets/2024/05/google-cloud-accidentally-nukes-customer-account-causes-two-weeks-of-downtime/) recently.
+
+- To mitigate threat 1, the new backup needs to have separate admin access.
+- To mitigate threat 2, the new backup needs to be in a separate cloud environment.
+
+## Implementation overview
+
+These new backups are hosted in a dedicated GCP account and have totally separate access controls compared to AWS.
+Specifically, none of the current `infra-admins` have admin access to this separate environment to protect against an account compromise.
+This GCP account is not used for anything else (just for backups).
+
+The backups are automatically copied daily by GCP.
+
+### Access 👤
+
+We limit admin access to the GCP backups to two members of the Rust Foundation for the following reasons:
+
+- _ensure a strong separation of access_: as explained in the first [motivation](#Motivation), the GCP admins should be different from the AWS admins.
+  This means we can't give admin access to any of the `infra-admins`.
+- _accountability_: the Rust Foundation employees are accountable for their actions, which means they can't delete the backups intentionally without breaking the law.
+
+People with admin access to the GCP account:
+
+- Joel Marcey (Director of Technology @ Rust Foundation)
+- Walter Pearce (Security Engineer @ Rust Foundation)
+
+People with read-only access to the GCP project:
+
+- `infra-admins`
+
+The admin access of the GCP account is bound to the `@rustfoundation.org` Google Workspace account.
+This means that if an employee leaves the Rust Foundation, they lose access to the GCP account.
+In this case we need to add a new admin to the GCP account.
+
+> [!NOTE]
+> The `infra-admins` team can have admin access to the GCP staging project if needed.
+
+### In case of emergency 🧯
+
+- In case our data in AWS is deleted, the `infra-admin` team can restore it by:
+  - copying the data from GCP to AWS using the GCP read-only access.
+  - restoring the `crates-io-index` bucket from the `db-dump` stored in the `crates-io` bucket. Use [this](https://github.com/rust-lang/crates.io/blob/e0bb0049daa12f5362def463b04febd6c036d315/src/worker/jobs/git.rs#L19-L129) code.
+- If the GCP synchronization mechanism breaks, the Infra team can raise a PR to fix the terraform configuration and a GCP admin can apply it.
```

Comment:
Nit: I'd either call the the "(I|i)nfrastructure team" (without a strong preference for capitalization) or the "infra-team", with a preference for the first. We're optimizing for readability here, and imo abbreviations make it more difficult to read long documents. But that's totally a personal opinion of mine...

Reply:
[82e3135](https://github.com/rust-lang/infra-team/pull/127/commits/82e31351f76a41b7ae3b5646a36ca26f3105ee53) 👍